src/main/kotlin/com/mineinabyss/idofront/serialization/BossBarSerializer.kt

```diff
@@ -0,0 +1,49 @@
+package com.mineinabyss.idofront.serialization
+
+import com.mineinabyss.idofront.messaging.miniMsg
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.descriptors.element
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.encoding.decodeStructure
+import kotlinx.serialization.encoding.encodeStructure
+import net.kyori.adventure.bossbar.BossBar
+import net.kyori.adventure.bossbar.BossBar.*
+
+object BossBarSerializer : KSerializer<BossBar> {
+
+    override val descriptor: SerialDescriptor = buildClassSerialDescriptor("BossBar") {
+        element<String>("name")
+        element<Float>("progress")
+        element<Color>("color")
+        element<Overlay>("overlay")
+    }
+
+    override fun serialize(encoder: Encoder, value: BossBar) =
+        encoder.encodeStructure(descriptor) {
+            encodeStringElement(descriptor, 0, value.name().examinableName())
```

Comment:
I dont think examinable name does what you want at all. I've pushed a new idofront version that lets you do Component.serialize() which should give you what you want here.